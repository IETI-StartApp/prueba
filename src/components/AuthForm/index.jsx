import React, {useRef, useState} from 'react';
import {useHistory} from "react-router";
import {Button, Checkbox, TextField, Typography} from "@material-ui/core";
import 'typeface-roboto';
import CircleChecked from '@material-ui/icons/CheckCircleOutline';
import CircleUnchecked from '@material-ui/icons/RadioButtonUnchecked';

import {CheckBoxArea, FooterWrapper, ForgotPasswdArea, InputArea, TextArea, Wrapper} from "./styles";
import google from '../../icons/google.svg'
import {Auth} from "../../authServices/firebase-config";
import {CustomTypography} from "../index";


export const AuthForm = ({functions}) => {
    const {signInWithEmailAndPassword, signInWithGoogle} = functions
    const [checked, setChecked] = useState(false);
    const emailRef = useRef('');
    const passwdRef = useRef('');
    const history = useHistory();

    function handleChange(event) {
        setChecked(event.target.checked)
    }

    const handleSubmit = async (e) => {
        e.preventDefault()
        try {
            console.log(emailRef.current.value, passwdRef.current.value)
            await signInWithEmailAndPassword(emailRef.current.value, passwdRef.current.value)
            if (Auth.currentUser) {
                history.push('/register-project')
            }
        } catch (error) {
            alert(error)
        }
    }

    const googleSignIn = async () => {
        await signInWithGoogle()
        Auth.currentUser ? history.push('/register-project') : history.push('/signin')
    }
    return (
        <Wrapper>
            <form onSubmit={handleSubmit}>
                <TextArea>
                    <Typography>
                        Email
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={emailRef} label="Email" variant="outlined" fullWidth type='email' required/>
                </InputArea>
                <TextArea>
                    <Typography>
                        Contrase単a
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={passwdRef} label='Contrase単a' variant="outlined" fullWidth type='password'
                               required/>
                </InputArea>
                <FooterWrapper>
                    <CheckBoxArea>
                        <Checkbox
                            className={`without-padding`}
                            checked={checked}
                            onChange={handleChange}
                            style={{
                                color: "#4A5568",
                                background: "none",
                                borderradius: "none"
                            }}
                            icon={<CircleUnchecked/>}
                            checkedIcon={<CircleChecked/>}
                        />
                        <CustomTypography text='#2D3748' style={{padding: '10px 0'}}>Recordarme</CustomTypography>
                    </CheckBoxArea>
                    <ForgotPasswdArea>
                        <CustomTypography
                            style={{padding: '10px 0', cursor: 'pointer'}}
                            text='#2C5282'
                            onClick={() => {
                                alert('You are going to be redirected!')
                            }}
                        >
                            多Olvidaste la contrase単a?
                        </CustomTypography>
                    </ForgotPasswdArea>
                </FooterWrapper>
                <Button
                    type='submit'
                    fullWidth
                    style={{
                        background: "#950740",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px'
                    }}>
                    Entrar
                </Button>
            </form>
            <Button onClick={googleSignIn}
                    fullWidth
                    style={{
                        background: "#2D3748",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px',
                        marginTop: '20px'
                    }}>
                <img src={google} height={20} width={20} style={{paddingRight: '11px'}} alt='logo'/>
                Entrar con Google
            </Button>
        </Wrapper>
    );

};
