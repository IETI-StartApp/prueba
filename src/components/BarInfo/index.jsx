import React from 'react';
import {styles} from './styles'
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
/*<BarInfo link ='Link de la pagina' pais = 'Pais del proyecto' texto ='descripcion  del proyecto'/>*/
export const BarInfo = ({
                          link,
                          pais,
                          texto,
}
) => {
  const classes = styles();
  return(
    <>
    <Paper className={classes.paper}>
      <Grid container spacing={0}>
        <Grid item xs = {2} >
          <a href = {link} className={classes.purpura}>Sitio web</a>
        </Grid>
        <Grid item xs= {2} className={classes.normal}>
          {pais}
        </Grid>
      </Grid>
    </Paper>
    <Grid container spacing= {0}  className={classes.paper}>
    <text className={classes.descripcion}>
      {texto}
    </text>
    </Grid>
    </>
  );
};

