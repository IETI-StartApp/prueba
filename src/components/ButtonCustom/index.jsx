import React from "react";

import {Wrapper} from "./styles";
import {Typography, withStyles} from "@material-ui/core";
/*
<Button green onClick={() => console.log("HOLA")} >
              Start Investing
            </Button>
 */
export const CustomTypography = withStyles({
    root: {
        color: props => props.text
    }
})(Typography);

export const Button = ({
                           children,
                           onClick,
                           color,
                           textColor,
                           widthFull
                       }) => {
    return (
        <Wrapper widthFull={widthFull} color={color} onClick={onClick}>
            <CustomTypography text={textColor}>
                {children}
            </CustomTypography>
        </Wrapper>
    );
};
