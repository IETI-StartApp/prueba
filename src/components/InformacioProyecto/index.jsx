import React from 'react';
import './styles.js'
import { Titulo, Emprendedor} from './styles'
import Grid from '@material-ui/core/Grid';
import { Avatar } from '@material-ui/core';

/*<InformacioProyecto nombreProyecto = "Nombre del proyecto" nombreUsuario = "Nombre del emprendedor" fotoPerfil = 'Ruta de la foto de perfil'/>*/
export const InformacioProyecto = ({
                                    fotoPerfil,
                                    nombreProyecto,
                                    nombreUsuario,
}) => {
  return(
    <div>
      <Grid container spacing={0} justify="center" alignItems="flex-start">
        <Grid item xs={1}>
          <Avatar variant = 'square'src={fotoPerfil}/> 
        </Grid>
        <Grid item xs={8}>
            <Titulo>
              <text>{nombreProyecto}</text>
            </Titulo>
        </Grid>
        <Grid item xs={1}/>
        <Grid item xs={8}>
          <Emprendedor>
            <text >{nombreUsuario}</text>
          </Emprendedor>
        </Grid>
      </Grid>
    </div>
  );
}

