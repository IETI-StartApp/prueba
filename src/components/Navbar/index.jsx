import React from 'react';
import {AppBar, Button, Hidden, IconButton, Toolbar, Typography} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import {useStyles} from './styles';
import {useAuth} from "../../authServices/Auth";

export default function Navbar({handleDrawerToggle}) {
    const classes = useStyles();
    const {signOut} = useAuth() || {};
    return (
        <AppBar position="static" className={classes.appBar}>
            <Toolbar>
                <IconButton
                    edge="start"
                    className={classes.menuButton}
                    color="inherit"
                    aria-label="menu"
                    onClick={() => handleDrawerToggle()}
                >
                    <MenuIcon/>
                </IconButton>
                <Typography variant="h6" className={classes.title}>
                    StartApp
                </Typography>
                <Hidden smDown>
                    <Button className={classes.buttonBar}>Opción 1</Button>
                    <Button className={classes.buttonBar}>Opción 2</Button>
                    <Button className={classes.buttonBar} variant="outlined" onClick={signOut}>Cerrar sesión
                    </Button>
                </Hidden>
            </Toolbar>
        </AppBar>
    );
}
