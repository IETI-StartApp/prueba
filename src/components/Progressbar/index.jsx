import React from 'react';
import {Step, StepLabel, Stepper, ThemeProvider} from '@material-ui/core';
import steps from './steps';
import progressTheme from './theme'


export const ProgressBar = ({activeStep}) => {
    return (

        <ThemeProvider theme={progressTheme}>
            <Stepper activeStep={activeStep} alternativeLabel>
                {steps.map(step => (
                    <Step key={step}>
                        <StepLabel>
                            {step}
                        </StepLabel>
                    </Step>
                ))}
            </Stepper>
        </ThemeProvider>
    )
}
