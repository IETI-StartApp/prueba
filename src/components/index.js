export * from './ButtonCustom'
export * from './AuthForm';
export * from './ProjectItems'
export * from './BarInfo'
export * from './InformacionFinazas'
export * from './InformacioProyecto'
export * from './Navbar'
export * from '../authServices/PrivateRoute'
export * from './Progressbar'
export * from './Register'
