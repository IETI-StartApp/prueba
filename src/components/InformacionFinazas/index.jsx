import { Grid, Paper, List, ListItemText } from '@material-ui/core';
import React from 'react';
import {styles} from './styles'
import { Button } from '../ButtonCustom';


export const InformacionFinazas = ({
                                    valor,
                                    inversionistas,
                                    valoracion,
                                    InversionMinima,
                                    socios,
                                    fechaInicio,
                                    FechaFin, 
}) => {
    const classes = styles();
    return(
        <>
        <Grid container >
            <Grid item xs={12} className={classes.valores}>
                {valor}
            </Grid>
        </Grid>
        <hr color='#960740' size ='3' width ='67%'/>
        <Paper className={classes.paper}>
            <Grid container >
                <Grid item xs={6}>
                    <List>
                        <ListItemText className={classes.numeros}>
                            <text className={classes.numeros}>{inversionistas}</text>     
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Inversionistas</text>
                        </ListItemText>
                    </List>
                </Grid>
                <Grid item xs={6}>
                    <List>
                        <ListItemText className={classes.numeros}>
                            <text className={classes.numeros}>{valoracion}</text>     
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Valoración</text>
                        </ListItemText>
                    </List>    
                </Grid>
            </Grid>
        </Paper>
        <Paper className={classes.paper}>
            <Grid container >
                <Grid item xs={6}>
                    <List>
                        <ListItemText className={classes.numeros}>
                            <text className={classes.numeros}>{InversionMinima}</text>     
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Inversión minima</text>
                        </ListItemText>
                    </List>
                </Grid>
                <Grid item xs={6}>
                    <List>
                        <ListItemText className={classes.numeros}>
                           <text className={classes.numeros}>{socios}</text>    
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Días faltantes</text>
                        </ListItemText>
                    </List>    
                </Grid>
            </Grid>
        </Paper>
        <Paper className={classes.paper}>
            <Grid container >
                <Grid item xs={6}>
                    <List>
                        <ListItemText >
                            <text className={classes.numeros}>{fechaInicio}</text>    
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Fecha inicio</text>
                        </ListItemText>
                    </List>
                </Grid>
                <Grid item xs={6}>
                    <List>
                        <ListItemText className={classes.numeros}>
                            <text className={classes.numeros}>{FechaFin}</text>
                        </ListItemText>
                        <ListItemText>
                            <text className={classes.descripcion}>Fecha Final</text>
                        </ListItemText>
                    </List>
                </Grid>
            </Grid>
        </Paper>

        <Grid item xs={12} className={classes.valores}>
            <Button green onClick={() => console.log("HOLA")} >
                Invertir ahora
            </Button>
        </Grid>
                        
                
        </>
    );
};