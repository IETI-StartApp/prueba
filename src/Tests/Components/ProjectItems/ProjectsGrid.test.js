import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import {ProjectsGrid} from "../../../components";

describe('Testing projects grid', () => {

    const wrapper = shallow(<ProjectsGrid/>)

    test('should load CardProject', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
