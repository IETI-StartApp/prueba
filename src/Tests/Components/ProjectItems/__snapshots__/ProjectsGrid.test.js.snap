// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing projects grid should load CardProject 1`] = `
<Fragment>
  <Navigation />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-cardGrid-4"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="1"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="2"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="3"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="4"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="5"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="6"
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
